api/tests/routes/pro/post_stocks_test.py

```diff
@@ -68,15 +71,46 @@ def test_edit_one_stock(self, mocked_async_index_offer_ids, client):
         # When
         stock_data = {
             "humanizedOfferId": humanize(offer.id),
-            "stocks": [{"humanizedId": humanize(existing_stock.id), "price": 20}],
+            "stocks": [{"humanizedId": humanize(existing_stock.id), "price": 20, "bookingLimitDatetime": None}],
         }
         response = client.with_session_auth("user@example.com").post("/stocks/bulk/", json=stock_data)
         created_stock = Stock.query.get(dehumanize(response.json["stockIds"][0]["id"]))
         assert offer.id == created_stock.offerId
         assert created_stock.price == 20
+        assert created_stock.bookingLimitDatetime == None
         assert len(Stock.query.all()) == 1
         mocked_async_index_offer_ids.assert_called_once_with([offer.id])
 
+    @patch("pcapi.core.search.async_index_offer_ids")
+    def test_edit_booking_limit_datetime(self, mocked_async_index_offer_ids, client):
+        booking_limit_datetime = datetime(2019, 2, 14)
+        new_booking_limit_datetime = datetime(2019, 2, 15)
+        offer = offers_factories.ThingOfferFactory(isActive=False, validation=OfferValidationStatus.DRAFT)
+        existing_stock = offers_factories.StockFactory(
+            offer=offer, price=10, bookingLimitDatetime=serialize(booking_limit_datetime)
+        )
+        offerers_factories.UserOffererFactory(
+            user__email="user@example.com",
+            offerer=offer.venue.managingOfferer,
+        )
+
+        # When
+        stock_data = {
+            "humanizedOfferId": humanize(offer.id),
+            "stocks": [
+                {
+                    "humanizedId": humanize(existing_stock.id),
+                    "price": 10,
+                    "bookingLimitDatetime": serialize(new_booking_limit_datetime),
+                }
+            ],
+        }
+        response = client.with_session_auth("user@example.com").post("/stocks/bulk/", json=stock_data)
+        created_stock = Stock.query.get(dehumanize(response.json["stockIds"][0]["id"]))
+        assert offer.id == created_stock.offerId
+        assert created_stock.bookingLimitDatetime == new_booking_limit_datetime
+        mocked_async_index_offer_ids.assert_called_once_with([offer.id])
+
```

Comment:
Ce test vérifie qu'on peut modifier la "booking limit datetime" quand elle existe : "valeur_non_nulle_avant" -> valeur_non_nulle_après". Mais le message de commit (et le ticket Jira) semble indiquer que le but de la modification, c'est d'autoriser à **supprimer** la "booking limit datetime" : "valeur_non_nulle_avant" -> None. Donc ce test ne vérifie pas le comportement modifié par cette pull request.

Au passage, je suggère un meilleur message de commit, qui n'indique pas que c'est spécifique à la v3 (car ça ne l'est pas) :  "fix: Allow to remove bookingLimitDatetime when editing non-event stock"

Reply:
Effectivement je n'ai pas vérifié le comportement dans ce test, je l'ai fait dans `test_edit_one_stock` car le test insérait déjà une valeur null avant la modification, et il n'y avait pas de test pour vérifier que l'utilisateur peut modifier une valeur qui était déjà non nulle. Je peux séparer les 3 comportements si nécéssaire. (edit one stock sans bookingLimitDatetime, edit avec valeur_non_nulle_avant --> null, valeur_non_nulle_avant -> valeur_non_nulle_après)